### PR TITLE
add release automation and offense language validation via github actions

### PR DESCRIPTION
**Pull request by ksimon1**
**What this PR does / why we need it**:
add release automation and offense language validation via github actions

This PR adds release automation, which creates new images and deploy
manifests when new github release is done.
Add script which checks offense language and if any is found, fail test.

**Release note**:
```
NONE
```
Signed-off-by: Karel Šimon <ksimon@redhat.com>

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+      - name: Get release
+        id: get_release
+        uses: bruceadams/get-release@v1.2.2
+        env:
+          GITHUB_TOKEN: ${{ secrets.ACTIONS_TOKEN }}
+
+      - name: Publish tekton images and generate release manifests
+        run: |
+          docker login -u="${{ secrets.QUAY_BOT }}" -p="${{ secrets.QUAY_PASSWORD }}" quay.io
```
Comment (atiratree) [inline .github/workflows/release.yaml:23]: can we use podman instead?

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      - name: Upload okd manifest
+        id: okd-manifest  
+        uses: actions/upload-release-asset@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.ACTIONS_TOKEN }}
+        with:
+          upload_url: ${{ steps.get_release.outputs.upload_url }}
+          asset_path: manifests/okd/kubevirt-tekton-tasks-okd.yaml
+          asset_name: kubevirt-tekton-tasks-okd.yaml
+          asset_content_type: text/plain
```
Comment (atiratree) [inline .github/workflows/release.yaml:47]: nit: newline

File: scripts/build-images.sh [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+REPO_DIR="$(realpath "${SCRIPT_DIR}/..")"
+
+source "${SCRIPT_DIR}/release-var.sh"
+source "${SCRIPT_DIR}/common.sh"
+
+
+visit "${REPO_DIR}"
+  visit modules
+    TASK_NAMES=(*)
+    for TASK_NAME in ${TASK_NAMES[*]}; do
```
Comment (atiratree) [inline scripts/build-images.sh:20]: not necessary: `*` should be enough

File: scripts/push-images.sh [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+SCRIPT_DIR="$(dirname "$(readlink -f "$0")")"
+REPO_DIR="$(realpath "${SCRIPT_DIR}/..")"
+
+source "${SCRIPT_DIR}/release-var.sh"
+source "${SCRIPT_DIR}/common.sh"
+
+visit "${REPO_DIR}"
+  visit modules
+    TASK_NAMES=(*)
+    for TASK_NAME in ${TASK_NAMES[*]}; do
```
Comment (atiratree) [inline scripts/push-images.sh:19]: same

File: makefile [hunk 5]
```diff
@@ -45,6 +45,14 @@ e2e-tests:
 onboard-new-task-with-ci-stub:
 	./scripts/onboard-new-task-with-ci-stub.sh
 
+build-images:
```
Comment (atiratree) [inline makefile:48]: maybe better to rename both to `build-release-images` ?

File: .github/workflows/release.yaml [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+      - name: Get release
+        id: get_release
+        uses: bruceadams/get-release@v1.2.2
+        env:
+          GITHUB_TOKEN: ${{ secrets.ACTIONS_TOKEN }}
+
+      - name: Publish tekton images and generate release manifests
+        run: |
+          echo ${{ secrets.QUAY_PASSWORD }} | podman login -u="${{ secrets.QUAY_BOT }}" --password-stdin quay.io
```
Comment (0xFelix) [inline .github/workflows/release.yaml:23]: Would `podman login -u "${{ secrets.QUAY_BOT }}" -p "${{ secrets.QUAY_PASSWORD }}" quay.io` work as well?

File: scripts/push-images.sh [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+            if [ ! -d  "${TASK_NAME}" ]; then
+                continue
+            fi
+            visit "${TASK_NAME}"
+                IMAGE_NAME_AND_TAG="tekton-task-${TASK_NAME}:${IMG_TAG}"
+                export IMAGE="${REGISTRY}/${REPOSITORY}/${IMAGE_NAME_AND_TAG}"
+
+                echo "Pushing ${IMAGE}"
+                
+                podman push "${IMAGE}" --tls-verify=false
```
Comment (0xFelix) [inline scripts/push-images.sh:29]: Why `--tls-verify=false`?

File: .github/workflows/release.yaml (hunk 6 above)
Comment (ksimon1) [reply .github/workflows/release.yaml:23]: yes, but it is insecure

File: scripts/push-images.sh (hunk 7 above)
Comment (ksimon1) [reply scripts/push-images.sh:29]: good catch :+1:

File: makefile [hunk 8]
```diff
@@ -45,6 +45,14 @@ e2e-tests:
 onboard-new-task-with-ci-stub:
 	./scripts/onboard-new-task-with-ci-stub.sh
 
+build-release-images:
+	./scripts/build-images.sh
```
Comment (atiratree) [inline makefile:49]: can we rename the files as well?

**Pull request merged**